app/src/main/java/pl/skipcode/kotlinbaseappmvvm/utils/network/rest/ResponseInterceptor.kt

```diff
@@ -0,0 +1,58 @@
+package pl.skipcode.kotlinbaseappmvvm.utils.network.rest
+
+import com.google.gson.JsonElement
+import com.google.gson.JsonParser
+import okhttp3.Interceptor
+import okhttp3.Response
+import pl.skipcode.kotlinbaseappmvvm.utils.configuration.ConfigurationInterface
+import java.lang.Exception
+import java.nio.charset.Charset
+import pl.skipcode.kotlinbaseappmvvm.utils.network.errors.ErrorStatus
+
+
+class ResponseInterceptor(
+        private val configuration: ConfigurationInterface
+) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+
+        val request = chain.request()
+        val response = chain.proceed(request)
+        val body = response.body()
+
+        var errorCode = 0
+        val errorStatus = response.code()
+
+        if (errorStatus >= 400) {
+
+            if (errorStatus == ErrorStatus.UNAUTHORIZED){
+                configuration.authorizationSubject.onNext(5)
+            }
+
+            // Only intercept JSON type responses and ignore the rest.
+            if (body?.contentType() != null && body.contentType()?.subtype() != null && body.contentType()?.subtype()?.toLowerCase().equals("json")) {
```

Comment:
To sie nadaje na jakiś poważny refaktor, poczytaj w necie o tym, pogoogluj w stylu "retrofit interceptor to handle api errors" czy coś, ja nie mam pomysłu na jakieś dobre rozwiązanie i nie mam już czasu dziś